src/com/rcv/Tabulator.java

```diff
@@ -254,14 +255,43 @@ private BigDecimal getThreshold(Map<String, BigDecimal> currentRoundCandidateToT
     return threshold;
   }
 
+  // purpose: determine if we should continue tabulating based on how many winners and losers
+  // have been selected and the configuration settings in use
+  // return: true if we should continue tabulating
+  private boolean shouldContinueTabulating() {
+    // how many candidates have already been eliminated
+    int eliminatedCandidates = candidateToRoundEliminated.keySet().size();
+//    // how many winners have been selected
+    int winners = winnerToRound.size();
+    if(config.continueTabulationTillAllCandidatesEliminated()) {
+      return (eliminatedCandidates + winners) < config.numCandidates();
```

Comment:
We want to go until there are two candidates left, not until everyone is eliminated.